### DISCUSSION TITLE
How to handle keepassxc attachments

### DISCUSSION THREAD

**Discussion by dboeckenhoff** [twpayne/chezmoi#2396] [Q&A] [answered]
Hey, AWESOME project! I have instantly migrated when I found it.
I am only stuck here: 

**How do I save a keepassxc attachment file (e.g. a ssh private key file to {{ .chezmoi.homeDir }}/.ssh/) to some directory?**

[Here](https://www.chezmoi.io/reference/templates/keepassxc-functions/keepassxcAttachment/) it says, the function "returns the attachment". I guess what I don't understand is **how** it returns it... (what type).

I would have expected a keepassxcSaveAttachementAt <key> <attachementName> <path> function.

I tried To add a file /home/dboe/.local/share/chezmoi/home/private_dot_ssh/private_id_ed25519.tmpl
with the following content:

`{{- keepassxcAttachment "ssh key" "id_ed25519" -}}
`
Which produces

```~/.local/share/chezmoi/home$ chezmoi apply
Invalid command attachment-export.
chezmoi: template: private_dot_ssh/private_id_ed25519.tmpl:1:4: executing "private_dot_ssh/private_id_ed25519.tmpl" at <keepassxcAttachment "ssh key" "id_ed25519">: error calling keepassxcAttachment: /usr/bin/keepassxc-cli attachment-export --stdout --quiet /home/IPP-HGW/dboe/share/Settings/keypass.kdbx 'ssh key' id_ed25519: exit status 1
Usage: keepassxc-cli [options] command
KeePassXC command line interface.

Available commands:
...
```
Happy for any help or examples.

**Comment by twpayne** [comment, marked as answer]
Thanks for the kind words!

`keepassAttachment` returns the contents of the attachment. You need `keepassxc-cli` version 2.7.0 or later for this though, and from the "Invalid command attachment-export" error message, I suspect that your version is older than this.

**Reply by twpayne** [reply]
I've added a check for this in #2397, which will give a clearer error message.

**Reply by dboeckenhoff** [reply]
Awesome! The great docu (and now I understand also suport) is one of the not to underestimate features of chezmoi
